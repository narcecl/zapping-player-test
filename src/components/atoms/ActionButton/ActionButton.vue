<script setup lang="ts">
import { computed } from 'vue';
import { RouterLink, type LocationAsRelativeRaw } from 'vue-router';

interface ActionButtonProps {
    to?: LocationAsRelativeRaw;
}

defineProps<ActionButtonProps>();

const componentClasses = computed(() => ({
    action_button: true,
}));
</script>

<template>
    <RouterLink v-if="to" :to="to" :class="componentClasses">
        <slot />
    </RouterLink>
    <button v-else :class="componentClasses">
        <slot />
    </button>
</template>

<style lang="scss">
.action_button {
    align-items: center;
    background-color: rgba(white, 0.1);
    border-radius: 100%;
    cursor: pointer;
    display: inline-flex;
    height: 52px;
    justify-content: center;
    transition: all 0.3s ease;
    width: 52px;

    &:hover {
        background-color: rgba(white, 0.2);
    }
}
</style>
