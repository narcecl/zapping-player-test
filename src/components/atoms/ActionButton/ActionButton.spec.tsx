import { describe, test, expect } from 'vitest';

import { mount } from '@vue/test-utils';
import ActionButton from './ActionButton.vue';

describe('ActionButton', () => {
    test('Default render (button)', () => {
        const wrapper = mount(<ActionButton>Hello Vitest</ActionButton>);
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.text()).toContain('Hello Vitest');
        expect([...wrapper.element.classList]).includes('action_button');
    });

    test('Default render (link)', () => {
        const wrapper = mount(<ActionButton>Hello Vitest</ActionButton>, {
            global: {
                stubs: {
                    RouterLink: {
                        template: '<a><slot /></a>',
                    },
                },
            },
            props: {
                to: { name: 'channel_view', params: { channelNumber: 1 } },
            },
        });
        expect(wrapper.html()).toMatchSnapshot();
        expect(wrapper.text()).toContain('Hello Vitest');
        expect(wrapper.element.tagName).toBe('A');
    });
});
