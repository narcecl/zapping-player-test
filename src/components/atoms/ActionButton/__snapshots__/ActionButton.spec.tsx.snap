// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ActionButton > Default render (button) 1`] = `"<button class="action_button">Hello Vitest</button>"`;

exports[`ActionButton > Default render (link) 1`] = `"<a to="[object Object]" class="action_button">Hello Vitest</a>"`;
